### ISSUE TITLE
[key-spacing] Option to enforce no newline between property key and value.

### ISSUE THREAD

**Issue by leonaves** [eslint/eslint#8500]
**Tell us about your environment**

* **ESLint Version:** 2.13.1
* **Node Version:** 6.9.5
* **npm Version:** 3.10.10

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**

```
    "key-spacing": [
      "error",
      { "mode": "strict", "beforeColon": false, "afterColon": true }
    ]
```

**What did you do? Please include the actual source code causing the issue.**

```js
cont myObj = {
  key:
  ['value'],
}
```

The above causes no error, which is expected; from the [documentation](http://eslint.org/docs/rules/key-spacing):

> In the case of long lines, it is acceptable to add a new line wherever whitespace is allowed.

However, I would appreciate an option for this rule to be able to enforce no new lines between the colon and the value, so that the following must be done:

```js
cont myObj = {
  key: [
    'value'
  ],
}
```

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
